src/xtgeo/grid3d/_grid_import_ecl.py

```diff
@@ -5,19 +5,27 @@
 
 import xtgeo
 from xtgeo.grid3d._egrid import EGrid, RockModel
-from xtgeo.grid3d._grdecl_grid import GrdeclGrid
+from xtgeo.grid3d._grdecl_grid import GrdeclGrid, GridRelative
 
 xtg = xtgeo.XTGeoDialog()
 
 logger = xtg.functionlogger(__name__)
 
 
-def import_ecl_egrid(self, gfile):
-    egrid = EGrid.from_file(gfile._file)
+def import_ecl_egrid(
+    self,
+    gfile,
+    units=None,
+    coordinates=GridRelative.MAP,
+    fileformat="egrid",
+):
+    egrid = EGrid.from_file(gfile._file, fileformat=fileformat)
+    if units is not None:
+        egrid.convert_units(units)
```

Comment:
Just to be clear, the `EGrid` when read will _always_ contain units (it's only the roff format that allows one to have dimensionless). We will fail on import if the file does not contain `GRIDUNIT`? (rather than failing when trying to convert something that isn't given...)

Also, when a user includes the fileformat and it is incorrect, then we abort mission, right? Stating that the file is in an incorrect format, rather than just falling back to guessing and continue?

As a user I would think that the units I provide in the `import_ecl_egrid` function must match whatever the units are in the stored file, not that the function itself will convert to whatever unit I ask for.. Could it be confusing for the user how this function works?

Reply:
Yes, importing EGRID without `GRIDUNIT` should give an error and aborting with incorrect format. `import_ecl_egrid`, with specifying units means converting to that unit. So `xtgeo.grid_from_file("my_grid.EGRID", units=Units.METRES)` will ensure that what you get is in metres regardless of what unit is in the file. Maybe we want that as a separate action in the public API? ie.
```
grd = xtgeo.grid_from_file("my_grid.EGRID")
grd.convert_units(xtgeo.Units.METRES)
```

Reply:
Personally I think so. It will be another step, but it will be less ambiguous :)

Reply:
After thinking about this I think I might have made a whoopsie. I am changing the behavior now though, but it is adding some complexity.

 The problem has to do with the interaction between `GRIDUNIT` , `MAPAXES` and `MAPUNITS`. `GRIDUNIT` gives the units of `COORD` and whether it is realtive to `MAPAXES` or not. `MAPUNITS` gives the units of `MAPAXES`. So one possible configuration is:
```
GRIDUNIT
 METRES MAP /
MAPUNITS
 FEET /
```

Reply:
Fixed it so that the idea is now to use `convert_units` to get your preferred units.